components/feed/LikeButton.tsx

```diff
@@ -1,60 +1,88 @@
 "use client";
 
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 
-const LikeButton = ({ filled = false }: { filled?: boolean }) => {
-  const [state, setState] = useState(filled);
+import { formatFollowers } from "@/lib/formatters";
+import isItemLiked from "@/lib/likeButton/isItemLiked";
+import likeItem from "@/lib/likeButton/likeItem";
+import unlikeItem from "@/lib/likeButton/unlikeItem";
+
+const LikeButton = ({
+  userId,
+  postId,
+  likes,
+}: {
+  userId: string;
+  postId: string;
+  likes: number;
+}) => {
+  const [liked, setLiked] = useState(false);
+
+  const [numLikes, setNumLikes] = useState(likes);
 
   useEffect(() => {
-    // eslint-disable-next-line no-console
-    console.log("like change");
-  }, [state]);
+    async function getLikeState() {
+      const likeState = await isItemLiked(userId, postId);
```

Comment:
We should try to avoid data fetching in the client — try finding wherever the post data is fetched and return the likes as a sub-array in that object.

Reply:
That also makes it so we dont have to use `useEffect` here — the likes will already be in the component